src/lj_gc.c

```diff
@@ -727,10 +727,14 @@ int LJ_FASTCALL lj_gc_step_jit(global_State *g, MSize steps)
   lua_State *L = gco2th(gcref(g->cur_L));
   L->base = tvref(G(L)->jit_base);
   L->top = curr_topL(L);
-  while (steps-- > 0 && lj_gc_step(L) == 0)
-    ;
-  /* Return 1 to force a trace exit. */
-  return (G(L)->gc.state == GCSatomic || G(L)->gc.state == GCSfinalize);
+  while (steps-- > 0 && lj_gc_step(L) == 0) {}
```

Comment:
any reason to use this form instead of the common "dangling semicolon" way?

Reply:
I wanted to make it more obvious the loop is for the side effects of the checks before it didn't have a semicolon or anything.

Reply:
ehh.... yes, it had a semicolon
```
    while (steps-- > 0 && lj_gc_step(L) == 0)
￼         ;
```
AFAICT, this is the common use of an empty loop.

Reply:
I must admit, I'm wondering if this is the sort of code that might be better expanded out...